pkg/config/autodiscover/autodiscover_targets.go

```diff
@@ -66,6 +70,55 @@ func FindTestTarget(labels []configsections.Label, target *configsections.TestTa
 	}
 
 	target.DeploymentsUnderTest = append(target.DeploymentsUnderTest, FindTestDeployments(labels, target, namespace)...)
+	target.Nodes = append(target.Nodes, GetNodesList()...)
+}
+
+// GetNodesList Function that return a list of node and what is the type of them.
+func GetNodesList() (nodes []configsections.Nodes) {
+	var nodeNames []string
+	var types []string
+	var Existinarray int = 0
+	context := common.GetContext()
+	tester := nodenames.NewNodeNames(common.DefaultTimeout, map[string]*string{"node-role.kubernetes.io/master": nil})
+	test, _ := tnf.NewTest(context.GetExpecter(), tester, []reel.Handler{tester}, context.GetErrorChannel())
+	_, err := test.Run()
+	if err != nil {
+		log.Error("Unable to get node list ", ". Error: ", err)
+	} else {
+		nodeNames = tester.GetNodeNames()
+		for i := range nodeNames {
+			node := configsections.Nodes{
+				Name: nodeNames[i],
+				Type: append(types, "master"),
+			}
+			nodes = append(nodes, node)
+		}
+	}
+	context = common.GetContext()
+	tester = nodenames.NewNodeNames(common.DefaultTimeout, map[string]*string{"node-role.kubernetes.io/worker": nil})
+	test, _ = tnf.NewTest(context.GetExpecter(), tester, []reel.Handler{tester}, context.GetErrorChannel())
+	_, err = test.Run()
+	if err != nil {
+		log.Error("Unable to get node list ", ". Error: ", err)
+	} else {
+		nodeNames = tester.GetNodeNames()
+		for i := range nodeNames {
+			node := configsections.Nodes{
+				Name: nodeNames[i],
+				Type: append(types, "worker"),
+			}
+			for j := range nodes {
+				if nodeNames[i] == nodes[j].Name {
+					nodes[j].Type = append(nodes[j].Type, "worker")
+					Existinarray = 1
+				}
+			}
+			if Existinarray == 0 {
+				nodes = append(nodes, node)
```

Comment:
```Existinarray``` should be set back to 0 
otherwise, you'll be always adding nodes to nodes

Also, I believe it's better to have a map